### PR TITLE
make split methode workable with Pattern class

### PR DESCRIPTION
**Pull request by ArekCzarnik**
Hi ... 

this is a simple change to make split workable with precompiled 'Regex' Patterns.

Thanks,
Arek

### CODE REVIEW

File: src/main/java/rx/observables/StringObservable.java [hunk 1]
```diff
@@ -1,12 +1,12 @@
 /**
  * Copyright 2014 Netflix, Inc.
- * 
+ * <p/>
```
Comment (akarnokd) [inline src/main/java/rx/observables/StringObservable.java]: Please don't change the license comment.

File: src/main/java/rx/observables/StringObservable.java [hunk 2]
```diff
@@ -357,23 +347,22 @@ public String call(Object obj) {
...
-     * 
+     *
      * See {@link Pattern}
      * <p>
      * <img width="640" src="https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/St.split.png" alt="">
-     * 
+     *
      * @param src
-     * @param regex
+     * @param pattern
```
Comment (akarnokd) [inline src/main/java/rx/observables/StringObservable.java]: Please add proper javadoc comments to the parameters.

File: src/main/java/rx/observables/StringObservable.java [hunk 3]
```diff
@@ -225,16 +220,13 @@ public boolean process(byte[] next, ByteBuffer last, boolean endOfInput) {
                                 bb.put(last);
                                 bb.put(next);
                                 bb.flip();
-                            }
-                            else { // next == null
+                            } else { // next == null
```
Comment (akarnokd) [inline src/main/java/rx/observables/StringObservable.java]: Please avoid whitespace and code style changes as much as possible; it makes the review more difficult. If you fix or introduce a method, you are free to code in your own style but don't change any unaffected code style.

Comment (ArekCzarnik) [inline src/main/java/rx/observables/StringObservable.java]: Yes you right, sorry intelliJ have make some magic foo after the commit.

**Comment by ArekCzarnik** [comment]
hope thats it... can you look on that?

File: src/main/java/rx/observables/StringObservable.java [hunk 4]
```diff
@@ -355,24 +355,49 @@ public String call(Object obj) {
         });
     }
 
-    /**
+ /**
```
Comment (akarnokd) [inline src/main/java/rx/observables/StringObservable.java]: Indentation missing here. Could you also squash the commits into just a single one?

**Comment by akarnokd** [comment]
Almost good. Could you squash the commits?

**Pull request closed without merging**